### DISCUSSION TITLE
Date range as a votable option

### DISCUSSION THREAD

**Discussion by sean-wils** [lukevella/rallly#171] [Ideas]
A popular request among my circle of friends is being able to vote for a date range (rather than a single day).

For example; choosing a date for a weekend away, a 2 week holiday, etc.

I've created some mockups to demonstrate what I mean:

![date_range_POC](https://user-images.githubusercontent.com/18518299/168491726-6b761d53-8a02-490c-90f3-c3c97ebd00e8.gif)

![date_range_POC-drag-datepicker](https://user-images.githubusercontent.com/18518299/168491775-851a2429-58d3-483a-be23-b880f1c41563.gif)

You could also still include single dates as well as ranged dates:

![date_range_POC-add-new-date](https://user-images.githubusercontent.com/18518299/168491849-32f58bd6-a779-4d93-b787-01bed4dd8836.gif)

I think it would be pretty intuitive on touchscreen devices too.

**Comment by lukevella** [comment]
Hey @sean-wils, nice work and appreciate you going through the effort of mocking it up. Is setting multiple options with the same start date possible/needed? If so how would that look like?

Also, one tricky thing about this is figuring out how this would translate to week view. Might be easier to just only allow this in month view and have the options reset if switching to week view.

**Reply by sean-wils** [reply]
Hey @lukevella, no problem!

You raise a good point about multiple options that share the same start dates, I hadn't considered that.

To avoid any additional complexity (or any major UI changes) I would probably say not to allow shared start/end dates between options. I personally I don't see much value in that.

It would make sense for dates to overlap however.
For example, choosing the best date for a 2 week holiday where the options are spaced out 1 week apart.

i.e.
1. 6th June - 19th June
2. 13th June - 26th June
3. 20th June - 3rd July

Again, I've created a few mockups to demonstrate how I could see that working:

![date_range_POC-overlap1](https://user-images.githubusercontent.com/18518299/168796461-3f978104-8bd7-4a7b-abe0-86af0ec9739e.gif)

![date_range_POC-overlap2](https://user-images.githubusercontent.com/18518299/168872418-bcaf75dd-a572-4102-bb02-fd9812c1a1a1.gif)

![date_range_POC-overlap3](https://user-images.githubusercontent.com/18518299/168871307-c6a9b33d-e4c2-4c68-8fae-415a64c7f5c6.gif)

There's obviously some logic you'd need to implement to ensure the feature doesn't get too complex/behaves itself.
I'd be happy to provide some scenarios if you'd want help fleshing that out!

**Reply by sean-wils** [reply]
>Also, one tricky thing about this is figuring out how this would translate to week view. Might be easier to just only allow this in month view and have the options reset if switching to week view.

I had some ideas on how that would look, but I agree with the idea to only allow this for Month view.

I think it would be a much lower priority/value feature for the Week view, and probably not worth the effort unless it was a popular enhancement request!

**Reply by lukevella** [reply]
Nice work! Can you let me know how you're creating these mock ups by the way? They are surprisingly detailed. I'll use these as a reference for this feature but can't say exactly when I'll be working on this. There are quite a few other things that are higher priority at the moment.

**Reply by sean-wils** [reply]
Cheers! Tbh it's just me playing around with the CSS using dev tools and stitching it together in Photoshop.
I'm not using any specific mockup tool.

>I'll use these as a reference for this feature but can't say exactly when I'll be working on this.

Not a problem. 👌  I just thought I'd raise the request and provide as much detail as I could. Glad it helped!

**Reply by kolaente** [reply]
What about only allowing this in month view and then expanding it later to month view? We probably don't need a version of it right at the start which can do everything, just something to iterate on.

**Comment by blackbyte-pl** [comment]
I would bump this up as a must have from my perspective :)

**Reply by lukevella** [reply]
Can you share what your use-case is for this feature that makes it a must have?

**Reply by herrrhein** [reply]
In the beginning of a new year I'm planning multiple events though the year with my friends. Basically we are searching for available prolonged weekends to go on a hiking trip or similar that - based on availability - will result either in a trip from Friday to Sunday or from Saturday to Monday.

**Comment by Epod** [comment]
I would be interested in this also

For example, we are planning a ski trip and this would be a range of dates the trip would be. We wanted to see if we could manage to head up Friday evening, or maybe we could only use a date range that started on Saturday.
